File: android_app_sample/android_sample/src/main/java/com/gaa/iap/sample/widget/LuckyNumberView.java
package com.gaa.iap.sample.widget;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.LinearLayout;

import androidx.constraintlayout.widget.ConstraintLayout;

import com.gaa.iap.sample.R;

import java.util.List;

public class LuckyNumberView extends ConstraintLayout {

    private LinearLayout mBallLayout;

    public LuckyNumberView(Context context) {
        super(context);
        init();
    }

    public LuckyNumberView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    private void init() {
        View view = LayoutInflater.from(getContext()).inflate(R.layout.view_lucky, this, true);
        mBallLayout = view.findViewById(R.id.balls);
        childInVisible();
    }

    public void setNumber(List<Integer> numbers) {
        int childCount = mBallLayout.getChildCount();
        if (numbers != null && numbers.size() == childCount) {
            for (int i = 0; i < childCount; i++) {
                BallView ball = (BallView) mBallLayout.getChildAt(i);
                ball.setNumber(numbers.get(i), true);
                ball.setVisibility(View.VISIBLE);
            }
        } else {
            childInVisible();
        }
    }

    private void childInVisible() {
        for (int i = 0; i < mBallLayout.getChildCount(); i++) {
            mBallLayout.getChildAt(i).setVisibility(View.INVISIBLE);
        }
    }
}

File: onestore_iap_sample/sample_luckyone/src/main/java/com/gaa/appdev/iap/sample/widget/LuckyNumberView.kt
package com.gaa.appdev.iap.sample.widget

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import androidx.constraintlayout.widget.ConstraintLayout
import com.gaa.appdev.iap.sample.R

class LuckyNumberView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyle: Int = 0)
    : ConstraintLayout(context, attrs, defStyle) {

    private var mBallLayout: LinearLayout

    init {
        val view: View = LayoutInflater.from(context).inflate(R.layout.view_lucky, this, true)
        mBallLayout = view.findViewById(R.id.balls)
        childInVisible()
    }

    fun setNumber(numbers: List<Int>) {
        val childCount = mBallLayout.childCount
        if (numbers.size == childCount) {
            for (i in 0 until childCount) {
                val ball: BallView = mBallLayout.getChildAt(i) as BallView
                ball.setNumber(numbers[i], true)
                ball.visibility = VISIBLE
            }
        } else {
            childInVisible()
        }
    }

    private fun childInVisible() {
        for (i in 0 until mBallLayout.childCount) {
            mBallLayout.getChildAt(i).visibility = INVISIBLE
        }
    }
}
